internal/util/aligned_ticker.go

```diff
@@ -0,0 +1,99 @@
+package util
+
+import (
+	"context"
+	"time"
+
+	"github.com/tilinna/clock"
+)
+
+// AlignedTicker is a wrapper around time.Ticker will align the time on a specific interval.  The first tick will
```

Comment:
Is the assumption here that the system clock is in sync with an NTP clock?

Reply:
Not really, there's nothing high precision here.  It's just about that sweet sweet xx:xx:x0 when you query the data.

I did consider fudging the timestamp (ie, round it up), rather than aligning the time, but that's bad for users, because they don't know which window they landed in, because they don't know the true window.